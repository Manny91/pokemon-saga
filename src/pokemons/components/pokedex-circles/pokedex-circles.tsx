import React from "react";
import styled from "styled-components";

type State = {
  handleClick?: () => void;
};
export const PokedexCircles = ({ handleClick }: State) => {
  return (
    <PokedexCirclesContainer onClick={handleClick}>
      <PokedexCircle />
      <PokedexCircleSmallRed />
      <PokedexCircleSmallYellow />
      <PokedexCircleSmallGreen />
    </PokedexCirclesContainer>
  );
};
function PokedexCircle() {
  return (
    <PokedexCircleStyled>
      <PokedexCircleInsideStyled />
    </PokedexCircleStyled>
  );
}
function PokedexCircleSmallRed() {
  return (
    <PokedexCircleSmallWrapperStyled>
      <PokedexCircleInsideRedStyled />
    </PokedexCircleSmallWrapperStyled>
  );
}
function PokedexCircleSmallYellow() {
  return (
    <PokedexCircleSmallWrapperStyled>
      <PokedexCircleInsideYellowStyled />
    </PokedexCircleSmallWrapperStyled>
  );
}
function PokedexCircleSmallGreen() {
  return (
    <PokedexCircleSmallWrapperStyled>
      <PokedexCircleInsideGreenStyled />
    </PokedexCircleSmallWrapperStyled>
  );
}

const PokedexCirclesContainer = styled.div`
  display: flex;
  cursor: pointer;
  border-bottom: 10px double ${props => props.theme.black};
  padding: 0px ${props => props.theme.spacing.sm};
`;
const PokedexCircleStyled = styled.div`
  width: 100px;
  height: 100px;
  background-color: ${props => props.theme.colors.white};
  border-radius: 50%;
  border: double black 10px;
  position: relative;

  height: 50px;
  width: 50px;
  margin: ${props => props.theme.spacing.sm};
  &:after {
    content: "";
    filter: blur(1px);
    position: absolute;
    top: 25%;
    width: 12%;
    height: 12%;
    background-color: ${props => props.theme.colors.white};
    left: 25%;
    border-radius: 50%;
  }
  @media ${props => props.theme.media.md} {
    margin: 1em;
    width: 100px;
    height: 100px;
  }
`;
const PokedexCircleSmallWrapperStyled = styled(PokedexCircleStyled)`
  height: 15px;
  width: 15px;
  @media ${props => props.theme.media.md} {
    width: 20px;
    height: 20px;
  }
`;
const PokedexCircleInsideStyled = styled.div`
  background-image: linear-gradient(
    to right,
    ${props => props.theme.colors.blue},
    ${props => props.theme.colors.darkBlue}
  );
  border-radius: 50%;
  width: 100%;
  height: 100%;
`;
const PokedexCircleInsideRedStyled = styled(PokedexCircleInsideStyled)`
  background-image: linear-gradient(
    to right,
    ${props => props.theme.colors.red},
    ${props => props.theme.colors.darkRed}
  );
`;
const PokedexCircleInsideYellowStyled = styled(PokedexCircleInsideStyled)`
  background-image: linear-gradient(
    to right,
    ${props => props.theme.colors.yellow},
    ${props => props.theme.colors.darkYellow}
  );
`;
const PokedexCircleInsideGreenStyled = styled(PokedexCircleInsideStyled)`
  background-image: linear-gradient(
    to right,
    ${props => props.theme.colors.green},
    ${props => props.theme.colors.darkGreen}
  );
`;
