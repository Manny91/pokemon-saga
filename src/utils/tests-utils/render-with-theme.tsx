import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
