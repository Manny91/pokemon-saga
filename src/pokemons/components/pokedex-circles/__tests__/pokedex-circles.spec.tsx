import React from "react";
import { PokedexCircles } from "../pokedex-circles";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokedexCircles Component", () => {
  it("should render properly", () => {
    const subject = renderWithTheme(<PokedexCircles />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
