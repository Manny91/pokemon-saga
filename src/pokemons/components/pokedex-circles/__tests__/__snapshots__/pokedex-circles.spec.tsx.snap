// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokedexCircles Component should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  border-bottom: 10px double;
  padding: 0px 10px;
}

.c1 {
  width: 100px;
  height: 100px;
  background-color: white;
  border-radius: 50%;
  border: double black 10px;
  position: relative;
  height: 50px;
  width: 50px;
  margin: 10px;
}

.c1:after {
  content: "";
  -webkit-filter: blur(1px);
  filter: blur(1px);
  position: absolute;
  top: 25%;
  width: 12%;
  height: 12%;
  background-color: white;
  left: 25%;
  border-radius: 50%;
}

.c3 {
  width: 100px;
  height: 100px;
  background-color: white;
  border-radius: 50%;
  border: double black 10px;
  position: relative;
  height: 50px;
  width: 50px;
  margin: 10px;
  height: 15px;
  width: 15px;
}

.c3:after {
  content: "";
  -webkit-filter: blur(1px);
  filter: blur(1px);
  position: absolute;
  top: 25%;
  width: 12%;
  height: 12%;
  background-color: white;
  left: 25%;
  border-radius: 50%;
}

.c2 {
  background-image: linear-gradient( to right, #6f95ff, #3c54a6 );
  border-radius: 50%;
  width: 100%;
  height: 100%;
}

.c4 {
  background-image: linear-gradient( to right, #6f95ff, #3c54a6 );
  border-radius: 50%;
  width: 100%;
  height: 100%;
  background-image: linear-gradient( to right, #e61515, #ce0000 );
}

.c5 {
  background-image: linear-gradient( to right, #6f95ff, #3c54a6 );
  border-radius: 50%;
  width: 100%;
  height: 100%;
  background-image: linear-gradient( to right, yellow, #dada03 );
}

.c6 {
  background-image: linear-gradient( to right, #6f95ff, #3c54a6 );
  border-radius: 50%;
  width: 100%;
  height: 100%;
  background-image: linear-gradient( to right, #008000, #016301 );
}

@media (min-width:768px) {
  .c1 {
    margin: 1em;
    width: 100px;
    height: 100px;
  }
}

@media (min-width:768px) {
  .c3 {
    margin: 1em;
    width: 100px;
    height: 100px;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 20px;
    height: 20px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c5"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c6"
      />
    </div>
  </div>
</div>
`;
